### DISCUSSION TITLE
Integration with the dashboard `Homepage`

### DISCUSSION THREAD

**Discussion by MDeLuise** [MDeLuise/plant-it#86] [Announcements]
Hi Plant-it Community!

I'm exploring the opportunity to integrate Plant-it with a popular dashboard called [homepage](https://github.com/gethomepage/homepage).
This integration comes with some great advantages:
* Seamless Integration: Having Plant-it alongside your other services on the dashboard for a unified experience.
* Widget Display: The ability to showcase Plant-it stats through a dedicated widget on the dashboard.

If you're keen on this integration, I'd love your support!
This request needs at least 10 upvotes to proceed with the integration, so please [upvote the request I created](https://github.com/gethomepage/homepage/discussions/2723).

Let's grow together! 🌱✨

**Comment by hunkyn** [comment]
seems like the widget is available. How do i get the API key for plant-it? I do not see that option to create an api.

**Reply by MDeLuise** [reply]
Hi,
You can create the API key using the REST API. Here’s how you can do it supposing that you have the deployment listening for port `8080`:
* go to `http://<server-url>:8080/api/swagger-ui/index.html`
* get the jwt token calling the `POST /authentication/login` with your credentials
* set the jwt token in swagger for the next calls (use the "authorize" button at the top)
* create the new api key calling `POST /api-key` optionally passing a name for the key

I realize that it would be useful to have the possibility of managing the API key directly from the app. I will implement this feature soon.
